### ISSUE TITLE
Does this support validation set while training to see generalization bound of the model?

### ISSUE THREAD

**Issue by erogol** [fchollet/keras#15]

**Comment by fchollet** [comment]
Sure! To monitor train loss and test loss after each batch, you could use:

```python
for e in range(nb_epoch):
   for batch_index in range(0, len(X_train)/batch_size+1):
      batch = range(batch_index*batch_size, min(len(X_train), (batch_index+1)*batch_size))
      train_loss = model.train(X_train[batch], y_train[batch])
      test_loss = model.evaluate(X_test, y_test, batch_size=batch_size)
```
Or alternatively, monitor train loss and test loss at each epoch (more practical):
```python
for e in range(nb_epoch):
    train_loss = model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=1)
    test_loss = model.evaluate(X_test, y_test, batch_size=batch_size)
```
